### DISCUSSION TITLE
Node LTS versions without codenames

### DISCUSSION THREAD

**Discussion by fgreinacher** [renovatebot/renovate#32334] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

We are using codenames to reference LTS Node versions. At the moment we're using Node 20 aka `Iron`. As Node 22 went LTS on [October 29](https://github.com/renovatebot/renovate/blob/main/data/node-js-schedule.json#L124) Renovate for a good reason wants to update to Node 22. The problem is that there was a small delay between Node 22 becoming LTS and Renovate [learning about the new codename](https://github.com/renovatebot/renovate/pull/32297). As codenames are considered version ranges Renovate made the following invalid bump:

<img width="1250" alt="image" src="https://github.com/user-attachments/assets/31c603a5-2527-451d-8e04-c9dd966fee22">

I'm not sure how we could prevent this. Happy about any guidance 🙇 

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Any idea when the codename was assigned? I was surprised it came late.

Also we maybe could try to prevent codename -> number upgrades regardless

**Reply by fgreinacher** [reply]
The Node codename was merged on October 29 (https://github.com/nodejs/Release/pull/1049).

The Renovate PR to update the schedule was created on Nov 3 (https://github.com/renovatebot/renovate/pull/32297) because the workflow is only running on Sundays (https://github.com/renovatebot/renovate/blob/main/.github/workflows/update-data.yml#L4).

It would certainly be possible to run that update workflow more often, but that would also not help much because not many Renovate instances are updated daily.

**Reply by fgreinacher** [reply]
I agree it would make sense to not upgrade from codename to number (and vice-versa).

If that's OK for you I can prepare a PR.

**Reply by rarkins** [reply]
Please do!
